### PR TITLE
fix(expressions): improved evaluateExpression api

### PR DESCRIPTION
**Pull request by cfieber**
* include error details on failure
* enable helper functions (allowUnknownKeys=true)

### CODE REVIEW

File: orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/TaskController.groovy [hunk 1]
```diff
@@ -294,9 +294,9 @@ class TaskController {
     def evaluated = contextParameterProcessor.process(
       [expression: expression],
       [execution: execution],
-      false
+      true
```
Comment (cfieber) [inline orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/TaskController.groovy]: `true` here to enable helper functions (`stage`, `deployedServerGroups`, etc)

Comment (christopherthielen) [reply orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/TaskController.groovy]: wonder why this was false to begin with?

Comment (robzienert) [reply orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/TaskController.groovy]: @jeyrschabu Probably knows

**Comment by tomaslin** [comment]
LGTM

File: orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/TaskController.groovy (hunk 1 above)
Comment (cfieber) [reply orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/TaskController.groovy]: @danielpeach Any concern with this? (you added it w/ false in #1336 - not sure what the exact use case was there)

Comment (danielpeach) [reply orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/TaskController.groovy]: nope, no memory of why though

**Pull request merged**